### DISCUSSION TITLE
Use CC2530 as switch module

### DISCUSSION THREAD

**Discussion by Flavien** [Koenkk/zigbee2mqtt#17093] [Devices]
Is it possible to turn one of these [CC2530 development boards](https://www.aliexpress.com/item/32330757866.html?pdp_npi=2%40dis%21EUR%21%E2%82%AC%204%2C99%21%E2%82%AC%203%2C74%21%21%21%21%21%402101d1bb16793587182528368e7d09%2157188812606%21btf&_t=pvid:05e8f54c-6025-4866-80e9-4262b64b709f&afTraceInfo=32330757866__pc__pcBridgePPC__xxxxxx__1679358718&spm=a2g0o.ppclist.product.mainProduct) into a switch module?

![image](https://user-images.githubusercontent.com/296540/226494531-6c7e1fe1-a8d7-4952-877e-b420074ae7e7.png)

I would like to connect some of the pins to the poles of a switch, and when closing the switch, the CC2530 would send a press action to Z2M.

Is it possible to achieve, and if so, what firmware and wiring diagram should I use?

**Comment by Danit2** [comment]
This firmware can do it:

https://ptvo.info

I love this prject and it works perfect with Z2M.

**Reply by Flavien** [reply]
Thanks, that looks great.
